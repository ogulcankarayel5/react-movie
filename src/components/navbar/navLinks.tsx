import React, { ReactElement } from 'react'
import {
  NavItem,
  NavLinksContainer,
  NavLinksRight,
} from 'components/navbar/style'
import { useAuth } from 'hooks'
import { BsFillBookmarkFill } from 'react-icons/bs'

type NavLinksProps = {
  onClose: () => void
}
export const NavLinks = ({ onClose }: NavLinksProps): ReactElement => {
  const { user } = useAuth()
  return (
    <NavLinksContainer>
      <NavItem
        activeStyle={{ color: '#CE1E37' }}
        to='/'
        onClick={onClose}
        exact
      >
        PREMIER
      </NavItem>
      <NavItem
        activeStyle={{ color: '#CE1E37' }}
        to='/films'
        onClick={onClose}
        isActive={(match, location) => {
          if (!match && location.pathname.includes('/movie')) {
            return true
          }

          if (!match) {
            return false
          }

          return true
        }}
      >
        FILMS
      </NavItem>
      <NavItem
        activeStyle={{ color: '#CE1E37' }}
        to='/tv'
        onClick={onClose}
        isActive={(match, location) => {
          if (!match && location.pathname.includes('/tv')) {
            return true
          }
          if (!match) {
            return false
          }

          return true
        }}
      >
        TV SERIES
      </NavItem>
      <NavLinksRight>
        <NavItem to='/favorites'>
          <BsFillBookmarkFill color='#CE1E37' />
        </NavItem>
        <NavItem
          activeStyle={{ color: '#CE1E37' }}
          to='/login'
          onClick={onClose}
        >
          {user ? user.displayName : 'LOGIN'}
        </NavItem>
      </NavLinksRight>
    </NavLinksContainer>
  )
}
