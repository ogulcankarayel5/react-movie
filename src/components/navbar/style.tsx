import { NavLink } from 'react-router-dom'
import styled, { css } from 'styled-components'
import { Text } from 'components'
import { breakPoints } from 'utils'

export const NavbarContainer = styled.div`
  background: transparent;
  height: 50px;
  position: absolute;
  top: 0;
  z-index: 10;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: space-evenly;
  padding: 0 100px;
`

export const NavLinksContainer = styled.div`
  display: flex;
  flex: 1;

  @media (max-width: ${breakPoints.lg}) {
    display: flex;
    flex-direction: column;
  }
`

export const NavLinksRight = styled.div`
  margin-left: auto;
  display: flex;
  flex-direction: row;

  @media (max-width: ${breakPoints.lg}) {
    margin-left: initial;
    flex-direction: column;
  }
`
export const LeftSection = styled.div`
  flex: 1;
`

export const RigtSection = styled.div<{ expanded: boolean }>`
  flex: 2;
  display: flex;
  justify-content: space-between;

  @media (max-width: ${breakPoints.lg}) {
    flex-direction: column;
    position: absolute;
    background-color: #101923;
    top: 0;
    width: 100%;
    height: 25vh;
    justify-content: center;
    align-items: center;
    transition: all 1s linear;
    opacity: 0;
    display: none;
    ${(props) =>
      props.expanded &&
      css`
        display: flex;
        opacity: 1;
      `}
  }
`

export const NavItem = styled(NavLink)`
  color: #cfd3d8;

  text-decoration: none;
  font-size: 1.2rem;
  &:not(:last-child) {
    margin-right: 20px;
  }

  @media (max-width: ${breakPoints.md}) {
    margin-top: 10px;
  }

  @media (max-width: ${breakPoints.xl}) {
    &:not(:last-child) {
      margin-right: 15px;
    }
  }
`

export const NavText = styled(Text)`
  color: #ededef;
  letter-spacing: 5px;
`

export const NavIconContainer = styled.div<{ expanded: boolean }>`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  position: absolute;
  right: 20px;
  & span:nth-child(1) {
    transform: rotate(0);
    ${(props) =>
      props.expanded &&
      css`
        transform: rotate(45deg);
      `}
  }

  & span:nth-child(2) {
    opacity: 1;
    transform: translateX(0);
    ${(props) =>
      props.expanded &&
      css`
        opacity: 0;
        transform: translateX(20px);
      `}
  }

  & span:nth-child(3) {
    transform: rotate(0);
    ${(props) =>
      props.expanded &&
      css`
        transform: rotate(-45deg);
      `}
  }
`

export const NavIconItem = styled.span`
  width: 25px;
  height: 3px;
  background-color: #ffffff;
  margin-top: 5px;
  transition: all 0.4s linear;
  transform-origin: 1px;
`
