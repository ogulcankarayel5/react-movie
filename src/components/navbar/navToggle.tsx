import React from 'react'
import { NavIconContainer, NavIconItem } from 'components/navbar/style'
type NavToggleProps = {
  expanded: boolean
  setExpanded: () => void
}

export const NavToggle = ({
  expanded,
  setExpanded,
}: NavToggleProps): React.ReactElement => {
  return (
    <NavIconContainer expanded={expanded} onClick={setExpanded}>
      <NavIconItem />
      <NavIconItem />
      <NavIconItem />
    </NavIconContainer>
  )
}
