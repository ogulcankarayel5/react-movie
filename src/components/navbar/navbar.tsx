import React, { ReactElement, useState } from 'react'
import {
  LeftSection,
  NavbarContainer,
  NavText,
  RigtSection,
} from 'components/navbar/style'
import { NavLinks } from 'components/navbar/navLinks'
import { NavToggle } from 'components/navbar/navToggle'
import { useMediaQuery } from 'react-responsive'

export const Navbar = (): ReactElement => {
  const [expanded, setExpanded] = useState(false)
  const isMobile = useMediaQuery({ maxWidth: 992 })

  return (
    <NavbarContainer>
      <LeftSection>
        <NavText bold text='MOVIES' />
      </LeftSection>

      <RigtSection expanded={expanded}>
        <NavLinks data-testid='nav-links' onClose={() => setExpanded(false)} />
      </RigtSection>
      {isMobile && (
        <NavToggle
          data-testid='toggle'
          expanded={expanded}
          setExpanded={() => setExpanded(!expanded)}
        />
      )}
    </NavbarContainer>
  )
}
