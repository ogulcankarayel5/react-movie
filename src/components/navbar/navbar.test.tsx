/* eslint-disable @typescript-eslint/no-empty-function */
import React from 'react'
import renderer from 'react-test-renderer'
import { render, fireEvent } from '@testing-library/react'
import { Navbar } from 'components/navbar/navbar'
import { MemoryRouter } from 'react-router-dom'
import { Context as ResponsiveContext } from 'react-responsive'

const routerWrapper = (comp: any) => {
  return <MemoryRouter>{comp}</MemoryRouter>
}

describe('snapshots', () => {
  test('should render correctly the navbar', () => {
    const tree = renderer.create(routerWrapper(<Navbar />)).toJSON()

    expect(tree).toMatchSnapshot()
  })
})

describe('component test', () => {
  test('should render the right items on larger than medium breakpoint device', () => {
    //not working as expected
    const { getByText, queryByTestId } = render(
      routerWrapper(
        <ResponsiveContext.Provider value={{ width: 768 }}>
          <Navbar />
        </ResponsiveContext.Provider>
      )
    )

    expect(getByText(/MOVIES/i)).toBeInTheDocument()

    expect(queryByTestId('toggle')).toBeNull()
  })

  test('it should render all nav items correctly', () => {
    const { getByText } = render(routerWrapper(<Navbar />))

    expect(getByText(/PREMIER/)).toBeInTheDocument()
    expect(getByText(/FILMS/)).toBeInTheDocument()
    expect(getByText(/TV SERIES/)).toBeInTheDocument()
    expect(getByText(/PREMIER/)).toBeInTheDocument()
    expect(getByText(/SHOW/)).toBeInTheDocument()
    expect(getByText(/LOGIN/)).toBeInTheDocument()
  })

  test('it should change the  active item when on click ', () => {
    const { getByText } = render(routerWrapper(<Navbar />))

    expect(getByText(/PREMIER/i)).toHaveStyle({ color: '#CE1E37' })

    fireEvent.click(getByText(/FILMS/i))

    expect(getByText(/PREMIER/i)).toHaveStyle({ color: '#cfd3d8' })
    expect(getByText(/FILMS/i)).toHaveStyle({ color: '#CE1E37' })
  })
})
