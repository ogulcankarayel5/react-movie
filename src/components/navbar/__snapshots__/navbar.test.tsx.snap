// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots should render correctly the navbar 1`] = `
<div
  className="sc-fFSPTT hnCUeK"
>
  <div
    className="sc-iemWCZ iXFNtQ"
  >
    <p
      className="sc-bdnxRM kPOorQ sc-dlMDgC jJMtGt"
      size="medium"
    >
      MOVIES
    </p>
  </div>
  <div
    className="sc-dIvrsQ ewjPVJ"
  >
    <div
      className="sc-bkbkJK bWSeVi"
    >
      <a
        aria-current="page"
        className="sc-hHEiqL iNSFzE active"
        href="/"
        onClick={[Function]}
        style={
          Object {
            "color": "#CE1E37",
          }
        }
      >
        PREMIER
      </a>
      <a
        aria-current={null}
        className="sc-hHEiqL iNSFzE"
        href="/films"
        onClick={[Function]}
      >
        FILMS
      </a>
      <a
        aria-current={null}
        className="sc-hHEiqL iNSFzE"
        href="/tv-series"
        onClick={[Function]}
      >
        TV SERIES
      </a>
      <a
        aria-current={null}
        className="sc-hHEiqL iNSFzE"
        href="/cartoons"
        onClick={[Function]}
      >
        CARTOONS
      </a>
      <a
        aria-current={null}
        className="sc-hHEiqL iNSFzE"
        href="/show"
        onClick={[Function]}
      >
        SHOW
      </a>
      <a
        aria-current={null}
        className="sc-hHEiqL iNSFzE"
        href="/sign-in"
        onClick={[Function]}
      >
        LOGIN
      </a>
    </div>
  </div>
</div>
`;
